### PR TITLE
EVG-15197  Don't error if child version doesn't exist when modifying

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-15197](https://jira.mongodb.org/browse/EVG-15197)

### Description 
This checks if a child patch is finalized before making modifications to avoid errors. 

### Testing 
Tested by creating a patch with a not finalized child patch in staging and changing the priority with no errors.

### CODE REVIEW

**Review by ablack12 — COMMENTED** [review]
How could the child patch not be finalized but the parent patch is finalized?

File: graphql/util.go [hunk 1]
```diff
@@ -767,10 +767,21 @@ func ModifyVersionHandler(ctx context.Context, dataConnector data.Connector, pat
...
+					p, err := patch.FindOneId(childPatchId)
 					if err != nil {
-						return errors.Wrap(mapHTTPStatusToGqlError(ctx, httpStatus, err), fmt.Sprintf("error modifying child patch '%s'", patchID))
+						return ResourceNotFound.Send(ctx, fmt.Sprintf("error finding child patch %s: %s", childPatchId, err.Error()))
 					}
+					if p == nil {
+						return ResourceNotFound.Send(ctx, fmt.Sprintf("child patch '%s' not found ", childPatchId))
+					}
+					// only modify the child patch if it is finalized
+					if p.Activated {
```
Comment (ablack12) [inline graphql/util.go:778]: more specifically what you're checking here is "does a version exist" which you can check by looking at p.Version instead

**Comment by malikchaya2** [comment]
> How could the child patch not be finalized but the parent patch is finalized?

Users can set it up for the child to wait on the parent. It can be helpful in cases where they may want to pass artifacts or values from the parent to the child.

File: graphql/util.go (hunk 1 above)
Comment (malikchaya2) [reply graphql/util.go:778]: Is that more fool-proof that checking the Activated field?

Comment (ablack12) [reply graphql/util.go:778]: Theoretically it's the same but just more general; the version definitely exists if the version field is set, whereas I don't remember if Activated can be changed. They probably both work though.

File: graphql/util.go [hunk 2]
```diff
@@ -767,10 +767,21 @@ func ModifyVersionHandler(ctx context.Context, dataConnector data.Connector, pat
 			}
 			if p.IsParent() {
 				for _, childPatchId := range p.Triggers.ChildPatches {
-					err = ModifyVersionHandler(ctx, dataConnector, childPatchId, modifications)
+					p, err := patch.FindOneId(childPatchId)
```
Comment (ablack12) [inline graphql/util.go:770]: can we get all of the patches with one query and then iterate through the patches instead?

**Review by ablack12 — APPROVED** [review]
LGTM, left a performance improvement suggestion but not blocking

File: graphql/util.go (hunk 2 above)
Comment (malikchaya2) [reply graphql/util.go:770]: That's a good point. There is another place where we do this so I will create another ticket to write a getchildPatches query and use it where appropriate.

File: graphql/util.go (hunk 1 above)
Comment (malikchaya2) [reply graphql/util.go:778]: checking p.version seems more accurate. I'll change it.

**Pull request merged**